SingularityService/src/main/java/com/hubspot/singularity/data/TaskManager.java

```diff
@@ -1379,24 +1397,35 @@ private void createTaskAndDeletePendingTaskPrivate(SingularityTask task)
     try {
       final String path = getTaskPath(task.getTaskId());
 
-      CuratorTransactionFinal transaction = curator
-        .inTransaction()
-        .create()
-        .forPath(path, taskTranscoder.toBytes(task))
-        .and();
-
-      transaction
-        .create()
-        .forPath(
-          getLastActiveTaskStatusPath(task.getTaskId()),
-          taskStatusTranscoder.toBytes(taskStatusHolder)
+      AtomicBoolean hasErr = new AtomicBoolean(false);
+      curator
+        .transaction()
+        .forOperations(
+          curator.transactionOp().create().forPath(path, taskTranscoder.toBytes(task)),
+          curator
+            .transactionOp()
+            .create()
+            .forPath(
+              getLastActiveTaskStatusPath(task.getTaskId()),
+              taskStatusTranscoder.toBytes(taskStatusHolder)
+            )
         )
-        .and()
-        .commit();
+        .forEach(
+          r -> {
+            if (r.getError() > 0) {
+              LOG.error("Error committing new task {} to zk {}", task.getTaskId(), r);
+              hasErr.set(true);
+            }
+          }
+        );
 
       // Not checking isActive here, already called within offer check flow
       leaderCache.putActiveTask(task.getTaskId());
       taskCache.set(path, task);
+      if (hasErr.get()) {
```

Comment:
should this be moved to a finally block, in case the transaction throws an exception?